### PR TITLE
[smart_vector] add destroy/clear functions for droppable T

### PR DESCRIPTION
**Pull request by lightmark**
### Description
drop/clear a smart vector if T has drop.

### Test Plan
ut

### CODE REVIEW

File: aptos-move/framework/aptos-stdlib/sources/data_structures/big_vector.move [hunk 1]
```diff
@@ -48,6 +48,18 @@ module aptos_std::big_vector {
         table_with_length::destroy_empty(buckets);
     }
 
+    /// Destroy the vector `v` if T has `drop`
+    public fun destroy<T: drop>(v: BigVector<T>) {
```
Comment (davidiw) [inline aptos-move/framework/aptos-stdlib/sources/data_structures/big_vector.move:52]: why not add the callback mechanism too?

**Review by davidiw — APPROVED** [review]

File: aptos-move/framework/aptos-stdlib/sources/data_structures/big_vector.move (hunk 1 above)
Comment (lightmark) [reply aptos-move/framework/aptos-stdlib/sources/data_structures/big_vector.move:52]: elaborate?

**Comment by lightmark** [comment]
@movekevin

File: aptos-move/framework/aptos-stdlib/sources/data_structures/smart_vector.move [hunk 2]
```diff
@@ -74,6 +74,16 @@ module aptos_std::smart_vector {
         option::destroy_none(big_vec);
     }
 
+    /// Destroy a table completely when T has `drop`.
+    public fun destroy<T: drop>(v: SmartVector<T>) {
```
Comment (movekevin) [inline aptos-move/framework/aptos-stdlib/sources/data_structures/smart_vector.move:78]: Is there a way to also add a clear() that wipes all data with a mutable reference? cc @banool

**Review by movekevin — APPROVED** [review]

**Pull request merged**